static/app/views/replays/detail/ai/ai.tsx

```diff
@@ -31,12 +33,37 @@ export default function Ai() {
 
 function AiContent() {
   const organization = useOrganization();
-  const {replay, aiSummaryContext} = useReplayContext(); // Data is queried in ReplayDetailsProviders.
+  const {replay, aiSummaryContext} = useReplayContext(); // Initial query is done in ReplayDetailsProviders.
   const replayRecord = replay?.getReplay();
   const project = useProjectFromId({project_id: replayRecord?.project_id});
 
   const openForm = useFeedbackForm();
 
+  // Until the first regenerate request, we use the prefetched result from aiSummaryContext.
+  const [regenerated, setRegenerated] = useState(false);
+  const regeneratedResults = useFetchReplaySummaryForceRegenerate({
```

Comment:
I would move this logic into the context, otherwise we  now two sources of truth for the replay summary. we want `aiSummaryContext` to always contain the latest summary.

Here's how I would structure it:

- create a `regenerateSummary` function that you expose via context
- call `regenerateSummary` when regenerate button is clicked
- @ryan953 is it safe to use the `regenerated` state like this in the hook? I'm concerned that other state changes in context can trigger `useQuery`. Maybe `useMutation()` is better here so that we have an imperative way to trigger this query on click?